### PR TITLE
feat(examples): add key expression argument to z_pub_thr and z_sub_thr

### LINKED ISSUE #2548: Throughput example should allow to specify the key expression
### Describe the feature

The `z_pub_thr` and `z_sub_thr` do not allow to specify the key expression. As a consequence it is not possible -- without changing the examples -- to test how a router throughput scales when running multiple instances using  same keys as opposed to running with different keys.

### PR DESCRIPTION
**Pull request by kydos**
## Description
Allow users to specify a custom key expression for the throughput examples via a CLI argument. This makes it possible to benchmark router throughput scaling across multiple instances using either shared or distinct key expressions without modifying the source.

### What does this PR do?
See above.

### Why is this change needed?
Because otherwise it is not possible to run throughput test on a user specified key-expr.

### Related Issues
Closes #2548

---
## 🏷️ Label-Based Checklist

Based on the labels applied to this PR, please complete these additional requirements:

**Labels:** `new feature`

## 🆕 New Feature Requirements

Since this PR adds a new feature:

- [ ] **Feature scope documented** - Clear description of what the feature does and why it's needed
- [ ] **Minimum necessary code** - Implementation is as simple as possible, doesn't overcomplicate the system
- [ ] **New APIs well-designed** - Public APIs are intuitive, consistent with existing APIs
- [ ] **Comprehensive tests** - All functionality is tested (happy path + edge cases + error cases)
- [ ] **Examples provided** - Usage examples in code comments or separate example files
- [ ] **Documentation added** - New docs explaining the feature, its use cases, and API
- [ ] **Feature flag considered** - Can the feature be enabled/disabled for gradual rollout?
- [ ] **Performance impact assessed** - Memory, CPU, storage implications measured
- [ ] **Integration tested** - Feature works with existing features

**Consider:** Can this feature be split into smaller, incremental PRs?

**Instructions:**
1. Check off items as you complete them (change `- [ ]` to `- [x]`)
2. The PR checklist CI will verify these are completed

*This checklist updates automatically when labels change, but preserves your checked boxes.*

### CODE REVIEW

File: examples/examples/z_sub_thr.rs [hunk 1]
```diff
@@ -71,24 +71,22 @@ fn main() {
...
             stats.increment();
-            if stats.finished_rounds >= m {
+            if stats.finished_rounds >= args.samples {
                 std::process::exit(0)
             }
         })
         .background()
         .wait()
-        .unwrap();
+        .expect("Failed to open Zenoh session");
```
[the remark this answers is not in the document]
Comment (fuzzypixelz) [reply examples/examples/z_sub_thr.rs:89]: @kydos Copilot is right here, but I would rather write "Failed to declare subscriber".
